.github/workflows/cygwin.yml

```diff
@@ -65,9 +65,21 @@ jobs:
       - name: Rebase NumPy compiled extensions
         run: |
           dash "tools/rebase_installed_dlls_cygwin.sh" 3.9
-      - name: Run NumPy test suite
-        run: >-
-          dash -c "/usr/bin/python3.9 runtests.py -n"
+      - name: Run NumPy test suite except F2PY
+        # There seem to be too many compiled extension modules.
+        # Let's try splitting off the tests by submodule to see if that helps.
+        # Not sure if that will run the top-level tests.
+        shell: "C:\\tools\\cygwin\\bin\\bash.exe -o igncr -eo pipefail {0}"
+        run: |
+          for submodule in array_api compat core distutils fft lib linalg ma matrixlib polynomial random tests typing;
```

Comment:
I am not sure what is the best path forward here:
- a list of directories to use, which will break the next time we add/remove a top-level directory
- some ugly path parsing like `find numpy -maxdepth 1 -type d | cut -f2 -d/ | grep -v f2py | grep -v numpy | grep -v "^_"` to get all the directories except `f2py`
- add a command line switch to `runtests.py` to allow skipping a subdirectory
- move the build to meson, and add a command line option to dev.py to skip a subdirectory

What you have is the simplest, but will be fragile. On the other hand, how often do we change the top-level directories?

Maybe is there a way to verify that your list is complete with a helper script?

Reply:
I think this might work `runtests.py ... -- --deselect f2py`

Reply:
```bash
for name in numpy/*; 
do 
    if [ -d "${name}" -a "${name}" != numpy/f2py -a "${name:6:1}" != "_" ] ;
    then
        echo ${name##numpy/}; 
    fi;
done
```
or
```bash
find numpy/* -maxdepth 0 -type d -a ! -name f2py -a ! -name _\* | cut -f2 -d/
```
would also work, but I don't think this separation is needed anymore.

Reply:
All submodule tests run at the same time again; PR title changed to match.